tests/test_md.py

```diff
@@ -0,0 +1,362 @@
+    single_point = SinglePoint(
+        struct_path=DATA_PATH / "NaCl.cif",
+        architecture="mace",
+        calc_kwargs={"model": MODEL_PATH},
+    )
+    dyn = ensemble(
+        struct=single_point.struct,
+    )
+    assert dyn.ensemble == expected
+
+
+def test_npt(tmp_path):
+    """Test NPT molecular dynamics."""
+    restart_stem = tmp_path / "res-Cl4Na4-npt"
+    restart_path_1 = tmp_path / "res-Cl4Na4-npt-300.0-2.xyz"
+    restart_path_2 = tmp_path / "res-Cl4Na4-npt-300.0-4.xyz"
+    traj_path = tmp_path / "Cl4Na4-npt-300.0-traj.xyz"
+    md_path = tmp_path / "Cl4Na4-npt-300.0-md.log"
+
+    single_point = SinglePoint(
+        struct_path=DATA_PATH / "NaCl.cif",
+        architecture="mace",
+        calc_kwargs={"model": MODEL_PATH},
+    )
+    npt = NPT(
+        struct=single_point.struct,
+        pressure=1.0,
+        temp=300.0,
+        steps=4,
+        traj_file=traj_path,
+        traj_every=1,
+        restart_stem=restart_stem,
+        restart_every=2,
+        md_file=md_path,
+        output_every=1,
+    )
+    npt.run()
+
+    restart_atoms_1 = read(restart_path_1)
+    assert isinstance(restart_atoms_1, Atoms)
+    restart_atoms_2 = read(restart_path_2)
+    assert isinstance(restart_atoms_2, Atoms)
+
+    traj = read(traj_path, index=":")
+    assert all(isinstance(image, Atoms) for image in traj)
+    assert len(traj) == 4
+
+    with open(md_path, encoding="utf8") as md_file:
+        lines = md_file.readlines()
+        assert "Pressure[bar] | T [K]" in lines[0]
+        assert len(lines) == 5
+
+
+def test_nvt_nh(tmp_path):
+    """Test NVT-Nosé–Hoover  molecular dynamics."""
+    restart_stem = tmp_path / "res-Cl4Na4-nvt-nh"
+    restart_path = tmp_path / "res-Cl4Na4-nvt-nh-300.0-3.xyz"
+    traj_path = tmp_path / "Cl4Na4-nvt-nh-300.0-traj.xyz"
+    md_path = tmp_path / "Cl4Na4-nvt-nh-300.0-md.log"
+
+    single_point = SinglePoint(
+        struct_path=DATA_PATH / "NaCl.cif",
+        architecture="mace",
+        calc_kwargs={"model": MODEL_PATH},
+    )
+    nvt_nh = NVT_NH(
+        struct=single_point.struct,
+        temp=300.0,
+        steps=3,
+        traj_file=traj_path,
+        traj_every=1,
+        restart_stem=restart_stem,
+        restart_every=3,
+        md_file=md_path,
+        output_every=1,
+    )
+    nvt_nh.run()
+
+    restart_atoms = read(restart_path)
+    assert isinstance(restart_atoms, Atoms)
+
+    traj = read(traj_path, index=":")
+    assert all(isinstance(image, Atoms) for image in traj)
+    assert len(traj) == 3
+
+    with open(md_path, encoding="utf8") as md_file:
+        lines = md_file.readlines()
+        assert " | T [K]" in lines[0]
+        assert len(lines) == 4
+
+
+def test_nve(tmp_path):
+    """Test NVE molecular dynamics."""
+    restart_stem = tmp_path / "res-Cl4Na4-nve"
+    restart_path = tmp_path / "res-Cl4Na4-nve-300.0-4.xyz"
+    traj_path = tmp_path / "Cl4Na4-nve-300.0-traj.xyz"
+    md_path = tmp_path / "Cl4Na4-nve-300.0-md.log"
```

Comment:
```suggestion
    md_path = tmp_path / "Cl4Na4-nve-T300.0-md.log"
```

Reply:
Do we want different defaults for the ensembles (other than just `self.ensemble`)?

At the moment we have

```python
if not self.md_file:
    self.md_file = f"{self.struct_name}-{self.ensemble}-{self.temp}-md.log"

if not self.traj_file:
    self.traj_file = f"{self.struct_name}-{self.ensemble}-{self.temp}-traj.xyz"

if not self.restart_stem:
    self.restart_stem = f"res-{self.struct_name}-{self.ensemble}"
```
etc. for all of them. Of course we can add "T" before `self.temp`, but should we instead define a different set of key parameters for each ensemble?

Reply:
if npt you want alse the pressure

Reply:
Anything else for the others?

Also, any reason not to include T in the restart files? If not, we can use the same label for all of them

Reply:
all files shall have it
also shall be -restart rather than -res

Reply:
the stem shall be the same and then prefix it with -stats.dat for -md.log, -traj and -restart
what was .out in poc shall be the .log but with no extra suffix

Reply:
> all files shall have it also shall be -restart rather than -res

Sorry, as in include pressure for non-NPT file names too?

Reply:
nope, nve,nvt will have T only npt, nph will have t and p